### PR TITLE
[WebAssembly] Add Support for Arc and Hyperbolic trig llvm intrinsics

### PR DESCRIPTION
**Pull request by farzonl**
## Change:
- WebAssemblyRuntimeLibcallSignatures.cpp: Expose the RTLIB's for use by WASM
-  Add trig specific test cases

## History
This change is part of an implementation of https://github.com/llvm/llvm-project/issues/87367's investigation on supporting IEEE math operations as intrinsics.
Which was discussed in this RFC:
https://discourse.llvm.org/t/rfc-all-the-math-intrinsics/78294

This change adds wasm lowering cases for `acos`, `asin`, `atan`, `cosh`, `sinh`, and `tanh`.

https://github.com/llvm/llvm-project/issues/70079
https://github.com/llvm/llvm-project/issues/70080
https://github.com/llvm/llvm-project/issues/70081
https://github.com/llvm/llvm-project/issues/70083
https://github.com/llvm/llvm-project/issues/70084
https://github.com/llvm/llvm-project/issues/95966

## Why Web Assembly?
From past changes to try and support  constraint intrinsics  the changes to the trig builtins to emit intrinsics\constraint intrinsics broke the WASM build. This is an attempt to preempt any such build break. 

- https://github.com/llvm/llvm-project/pull/95082
- https://github.com/llvm/llvm-project/pull/94559#issuecomment-2159923215

### CODE REVIEW

File: llvm/test/CodeGen/WebAssembly/libcalls-trig.ll [hunk 1]
```diff
@@ -0,0 +1,296 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc < %s -disable-wasm-fallthrough-return-opt -wasm-keep-registers | FileCheck %s
+
+; Test a subset of compiler-rt/libm libcalls expected to be emitted by the wasm backend
+
+target triple = "wasm32-unknown-unknown"
+
+declare fp128 @llvm.sin.f128(fp128)
+declare fp128 @llvm.cos.f128(fp128)
```
Comment (farzonl) [inline llvm/test/CodeGen/WebAssembly/libcalls-trig.ll:9]: There were no existing f128 cases for sin\cos\tan so added them even though this change does not add them.

File: llvm/test/CodeGen/WebAssembly/libcalls-trig.ll [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+declare double @llvm.cos.f64(double)
+declare double @llvm.tan.f64(double)
+declare double @llvm.asin.f64(double)
+declare double @llvm.acos.f64(double)
+declare double @llvm.atan.f64(double)
+declare double @llvm.sinh.f64(double)
+declare double @llvm.cosh.f64(double)
+declare double @llvm.tanh.f64(double)
+
+declare float @llvm.sin.f32(float)
```
Comment (farzonl) [inline llvm/test/CodeGen/WebAssembly/libcalls-trig.ll:28]: There were no f32 cases for  sin\cos\tan so added them.

File: llvm/test/CodeGen/WebAssembly/libcalls-trig.ll [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+declare fp128 @llvm.tan.f128(fp128)
+declare fp128 @llvm.asin.f128(fp128)
+declare fp128 @llvm.acos.f128(fp128)
+declare fp128 @llvm.atan.f128.i32(fp128)
+declare fp128 @llvm.sinh.f128(fp128)
+declare fp128 @llvm.cosh.f128(fp128)
+declare fp128 @llvm.tanh.f128(fp128)
+
+declare double @llvm.sin.f64(double)
+declare double @llvm.cos.f64(double)
```
Comment (farzonl) [inline llvm/test/CodeGen/WebAssembly/libcalls-trig.ll:19]: There were existing f64 cases for cos\tan but for consistencly included them here aswell.

**Review by dschuff — APPROVED** [review]
Thanks!

**Pull request merged**